### ISSUE TITLE
Dynamic segments?

### ISSUE THREAD

**Issue by dflock** [brujoand/sbp#14]
Having to update `helpers/imports.bash` when you create a new segment is non-obvious. It seems like this could be avoided by doing something like this at the bottom of `imports.bash`, instead of importing each segment individually by filename:

``` bash
...
#
# # shellcheck source=segments/timestamp.bash
# source "$sbp_path"/segments/timestamp.bash

dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

for f in $dir/../segments/*.bash; do
  source $f;
done
```

This would allow you to just drop the new segment into the `segments` folder and it would get imported automatically. I'm using the above code and it works for me.

The only downsides I can see for this are:
- I don't think the `shellcheck source...` directive can be made to work with this
- People would have to be careful not to put other stuff in the `segments` folder - helper scripts, etc...

**Comment by brujoand** [comment]
Yeah, this has been bothering me a lot. Initially I did just iterate over all the subfolders and sourced all .bash files. The advantages is as you point out that the import.bash file isn't very intuitive, but sourcing all files also seems like a brute approach. And I haven't been able to get shellcheck to go along with this. 

I think that it would be a good idea to always manually source files from helpers/ and then automatically source just the enabled segments and hooks. In which case shellcheck would be somewhat confused, but we wouldn't be sourcing files unintentionally, and the imports.bash would be gone. Any thoughts on that?

**Comment by dflock** [comment]
That sounds like a good idea - use `_sbp_settings_segments_*` to figure out which files to import - and you could look in both the `sbp/segments` folder, as well as `~/.config/sbp/segments`, maybe, as suggested here: https://github.com/brujoand/sbp/issues/17 ?

**Comment by brujoand** [comment]
Aye, I think that would work out nicely with the local segments/hooks.

**Comment by brujoand** [comment]
Implemented by #64
